typesetters/base.lua

```diff
@@ -145,6 +145,13 @@ function typesetter.declareSettings(_)
     default = false,
     help = "When true, a warning is issued when a soft hyphen is encountered"
   })
+
+  SILE.settings:declare({
+    parameter = "typesetter.dialogue",
```

Comment:
Something about this setting name doesn't quite sit right with me—possible just the lack of scope. I don't have a great suggestion though. What about `typesetter.em-dash-demarcates-dialogue` or something else similarly verbose that gives some more clues about what is being changed? just "dialogue" sounds more to me like an environment name where any number of settings could have been adjusted than a single setting.

Reply:
I am not happy with `typesetter.dialogue` either :)
Erm, perhaps `typesetter.emdashSpeakerChange` simply?

Reply:
Or possibly very explicit: `typesetter.fixedSpacingAfterInitialEmdash` ?